packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -1140,10 +1138,20 @@ class FlutterPlugin implements Plugin<Project> {
                             return "lib/${abi}/lib${filename}"
                         }
                     }
+                    // Copy the native assets created by build.dart and placed in build/native_assets by flutter assemble.
+                    // The `$project.buildDir` is '.android/Flutter/build/' instead of 'build/'.
+                    def buildDir = "${getFlutterSourceDirectory()}/build"
+                    def nativeAssetsDir = "${buildDir}/native_assets/android/jniLibs/lib"
+                    from("${nativeAssetsDir}/${abi}") {
+                        include "*.so"
+                        rename { String filename ->
+                            return "lib/${abi}/${filename}"
```

Comment:
Small nit:

Tasks in Gradle should not directly access (e.g. copy, move) output artifacts of other tasks. This is what's happening here: the `packJniLibsTask` directly accesses `*.so` files in `build/native_assets/android/jniLibs/lib` (and thus depends on whatever task created that directory).

Instead, if you want to make task A depend on task B, you should use implicit dependencies between tasks and let Gradle figure out the dependency graph for you. In other words, using `dependsOn` is usually a code smell, as seen above in line 1131.

I find that [this blogpost from a former Gradle engineer](https://melix.github.io/blog/2021/10/gradle-quickie-dependson.html) explains this topic well.

This is a nit and not directly related to this PR at all, but I just wanna raise awareness of this issue - it occurs in almost all tasks of Flutter Gradle Plugin. A while back I created https://github.com/flutter/flutter/issues/123934 to track this issue.

Reply:
Ack!